scopes/react/react/react.docs.md

```diff
@@ -0,0 +1,422 @@
+
+For example:
+
+```ts
+// ...
+
+// Import the task
+import { CustomTask } from './custom.task';
+
+export class CustomReact {
+  // ...
+
+  static async provider([envs, react]: [EnvsMain, ReactMain]) {
+    // Get the environment's default build pipeline using the 'getBuildPipe' service handler
+    const reactPipe = react.env.getBuildPipe();
+
+    // Add the custom task to the end of the build tasks sequence.
+    const tasks = [...reactPipe, new CustomTask()];
+
+    const newReactEnv = react.compose([react.overrideBuildPipe(tasks)]);
+
+    // ...
+  }
+}
+```
+
+#### overrideDependencies
+
+```ts
+overrideDependencies(dependencyPolicy: DependenciesPolicy): EnvTransformer
+```
+
+This method receives a Bit dependency-policy object. It overrides the default dependency policy for components using this environment.
+
+Each key-value pair in a dependency-policy object signifies the package and the version to be used. It also uses the `-` notation to signify a module should not be defined as a dependency of a certain type (dev, peer or standard).
+
+For example:
+
+```js
+// ...
+
+const newDependencies = {
+      dependencies: {
+        react: '-',
+      },
+      devDependencies: {
+        '@types/react': '16.9.43',
+        '@types/jest': '~26.0.9',
+        '@types/mocha': '-',
+        '@types/react-router-dom': '^5.1.5',
+      },
+      peerDependencies: {
+        react: '^16.13.1' || this.config.reactVersion,
+        'react-dom': '^16.13.1',
+      },
+    };
+}
+
+export class CustomReact {
+
+  // ...
+
+  static async provider([envs, react]: [EnvsMain, ReactMain]) {
+
+    const newReactEnv = react.compose([
+      react.overrideDependencies(newDependencies)
+    ]);
+
+    // ...
+
+  }
+}
+```
+
+> The above example shows the 'react' library being removed as a (runtime) dependency and added as a peer dependency.
+
+
+##### Handling multiple peer dependencies in a workspace with multiple environments
+
+Peer dependencies assume a single “hosting code”, a single application. A Bit workspace is not a single “host”.
+It may create multiple “hosts” as each "application" is generated by a different Bit environment (that's what gives us the freedom to author and explore components of all sorts of types in one single workspace).
+That means different "hosts", different environments, may have the same package defined as a peer dependency only with different versions. 
+To prevent such conflicts, set the `resolveFromEnv` property of that dependency as `true`. This will make sure to install the needed peer dependency in a directory set by the environment.
+It will then resolve the path to that installed package for each component requiring it. Each component will only use the package relevant to its environment.
```

Comment:
mention that it will uses symlink to the env location, so you still get a single instance

Reply:
Done.